### PR TITLE
updating button tests

### PR DESCRIPTION
**Pull request by hsbacot**
#### Related issues
Fixes https://meetup.atlassian.net/browse/SDS-904

#### Description
Updating failing tests related to `<Button />`

### CODE REVIEW

File: src/forms/button.test.jsx [hunk 1]
```diff
@@ -18,83 +12,62 @@ describe('Button', () => {
 		});
 	});
 
-	it('applies variant classes for each variant prop', () => {
-		const variants = ['primary', 'fullWidth', 'small', 'bordered'];
+	it('has a default state', () => {
+		const button = mount(<Button>Click Me</Button>);
```
Comment (sadafie) [inline src/forms/button.test.jsx]: Slightly more DRY approach would be to define a `renderComponent` function:

```
const renderComponent = (props = {}) => mount(<Button {...props}>Click Me</Button>);

describe('Button', () => {
...

   const button = renderComponent({primary:true});
})
```

File: src/forms/Button.jsx [hunk 2]
```diff
@@ -22,17 +22,17 @@ class Button extends React.PureComponent {
 		// since this functionality is built into button in order to enforce strict sizes on button icons
 		const props = { icon: icon && icon.props && icon.props.shape };
 
-		if (component !== 'button') {
+		if (component !== 'button' || 'a') {
```
Comment (jaclynj) [inline src/forms/Button.jsx]: This logic isn't right. the second half of the statement is just 'a' which will always evaluate to true. I'm not sure what you actually want here.. maybe this `component !== 'button' && component !== 'a'`

Comment (jaclynj) [reply src/forms/Button.jsx]: I think you should also add a test to make sure this warning is logged only when you want it to.

File: src/forms/button.test.jsx [hunk 3]
```diff
@@ -18,83 +12,62 @@ describe('Button', () => {
...
-		button.simulate('click');
-		expect(onClick).toHaveBeenCalled();
+	it('has a bordered state', () => {
+		const button = mount(<Button bordered>Click Me</Button>);
+		expect(
+			button.getDOMNode().attributes.getNamedItem('data-swarm-button').value
+		).toBe('bordered');
 	});
 
-	it('does not execute onClick when disabled', () => {
```
Comment (jaclynj) [inline src/forms/button.test.jsx]: Should we also have this test? make sure not to call onclick when the button is disabled?

Comment (hsbacot) [reply src/forms/button.test.jsx]: This is a test that predates the support being default in react 🦖

File: src/forms/button.test.jsx (hunk 1 above)
Comment (hsbacot) [reply src/forms/button.test.jsx]: I prefer a little repetition here to keep the react api in the test definition. This prevents looking around for the definition of the `renderComponent` when you want to change.

**Pull request merged**